test/antrea/k8spolicy.dl

```diff
@@ -0,0 +1,346 @@
+                            }
+                        },
+	                    LabelSelectorOpExists -> map_contains_key(labels, e.reqkey),
+	                    LabelSelectorOpDoesNotExist -> not map_contains_key(labels, e.reqkey)
+                    })
+                };
+                matches
+            }
+        }
+    }
+}
+
+// A label selector requirement is a selector that contains values, a key, and an operator that
+// relates the key and values.
+typedef LabelSelectorRequirement = LabelSelectorRequirement {
+	// key is the label key that the selector applies to.
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	reqkey: string,
+	// operator represents a key's relationship to a set of values.
+	// Valid operators are In, NotIn, Exists and DoesNotExist.
+	operator: LabelSelectorOperator,
+	// values is an array of string values. If the operator is In or NotIn,
+	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the values array must be empty. This array is replaced during a strategic
+	// merge patch.
+	// +optional
+	values: Vec<string>
+}
+
+// A label selector operator is the set of operators that can be used in a selector requirement.
+typedef LabelSelectorOperator = LabelSelectorOpIn
+	                          | LabelSelectorOpNotIn
+	                          | LabelSelectorOpExists
+	                          | LabelSelectorOpDoesNotExist
+
+function labelSelectorOperator2string(op: LabelSelectorOperator): string =
+{
+    match (op) {
+        LabelSelectorOpIn           -> "In",
+	    LabelSelectorOpNotIn        -> "NotIn",
+	    LabelSelectorOpExists       -> "Exists",
+	    LabelSelectorOpDoesNotExist -> "DoesNotExist"
+    }
+}
+
+typedef PolicyType = PolicyTypeIngress
+                   | PolicyTypeEgress
+
+// NetworkPolicySpec provides the specification of a NetworkPolicy
+typedef NetworkPolicySpec = NetworkPolicySpec {
+	// Selects the pods to which this NetworkPolicy object applies. The array of
+	// ingress rules is applied to any pods selected by this field. Multiple network
+	// policies can select the same set of pods. In this case, the ingress rules for
+	// each are combined additively. This field is NOT optional and follows standard
+	// label selector semantics. An empty podSelector matches all pods in this
+	// namespace.
+	podSelector: Ref<LabelSelector>,
+
+	// List of ingress rules to be applied to the selected pods. Traffic is allowed to
+	// a pod if there are no NetworkPolicies selecting the pod
+	// (and cluster policy otherwise allows the traffic), OR if the traffic source is
+	// the pod's local node, OR if the traffic matches at least one ingress rule
+	// across all of the NetworkPolicy objects whose podSelector matches the pod. If
+	// this field is empty then this NetworkPolicy does not allow any traffic (and serves
+	// solely to ensure that the pods it selects are isolated by default)
+	// +optional
```

Comment:
I think this means that it can be `nil`, which normally translates into `Option<>` in DDlog, but in case of vector `nil` is equivalent to an empty vector, so no need for `Option`.